### DISCUSSION TITLE
v3.0.154.0

### DISCUSSION THREAD

**Discussion by doug24** [dnGrep/dnGrep#755] [General]
[![v3.0.154.0](https://img.shields.io/github/downloads/dnGrep/dnGrep/v3.0.154.0/total.svg)](https://github.com/dnGrep/dnGrep/releases/tag/v3.0.154.0)

## Changes in this release:

**Locked file check**: the previous release added a check to see if files are locked by another application when checking for read-only files.  This check can be very slow and made dnGrep unresponsive at times.  This locked file check has been removed and reverted to the previous error handling for locked files.  

**#717 Settings file format**: In the dnGREP.Settings.dat file, data that was previous saved as binary data is now all plain text. Your existing settings file will be converted to the new format on first use.  

**#745 Preview window clear**: When all files have been removed from the results list, the Preview window is now cleared.

**#746 Everything search path**: When in Everything search mode, the path is now enclosed in quotes when the path contains spaces. This applies when starting dnGrep from File Explorer or command line, using the file/folder selection dialog, drag and drop a folder on the dnGrep window, and when changing from Regular Expression or Wildcard search mode.

**Added français translation.** Thank-you for the new translation!

You are welcome and encouraged to add new translations or improve any of the existing translations, see [Translate dnGrep](https://github.com/dnGrep/dnGrep/wiki/Translate-dnGrep) for more information. Do as much or as little as you like: there are some translations missing just a few phrases, and some new languages that need to be more complete before being included in a dnGrep release. 

**Future release** I am thinking about adding scripting in a future release, maybe sometime this winter when I have more time.  Click the discussion link below for more information or if you want to contribute your thoughts.

### Checksum information

Name: dnGREP.3.0.154.x64.msi
SHA256: deb58fd5fd81651e1f8fb2ce245fd7d47df9482c508cee1cf18f3d90a416a210

Name: dnGREP.3.0.154.x86.msi
SHA256: 5ea2666d4df1dccd918ac1773cd7f37e20cf67d75273a737db9675b916989084

Free code signing provided by [SignPath.io](https://signpath.io/), certificate by [SignPath Foundation](https://signpath.org/).

<hr /><em>This discussion was created from the release <a href='https://github.com/dnGrep/dnGrep/releases/tag/v3.0.154.0'>v3.0.154.0</a>.</em>

**Comment by doug24** [comment]
I am thinking about adding scripting in a future release.  The scripts could be run from a new menu on the main window or command line.  Scripting could include:

- Setting parameters - anything on the main window, or anything in the dnGrep.Settings.dat file 
- Applying a bookmark
- Running Search
- Doing Replace
- Generating a report
- Copy, Move, and Delete files

I'm thinking a script would be a plain text file, with one script per file and with one command per line within the file.

**Comment by jcmil101** [comment]
I would love to be able to do SAR using delimited literal text in non-overlapping search.
Having the ability to use an input & output list to automate multiple searches would be awesome.
